security/pkg/caclient/client.go

```diff
@@ -58,14 +55,13 @@ func (c *CAClient) Retrieve(options *pkiutil.CertOptions) (newCert []byte, certC
 	retries := 0
 	retrialInterval := c.initialRetrialInterval
 	for {
-		privateKey, req, reqErr := c.CreateCSRRequest(options)
+		privateKey, req, reqErr := c.createCSRRequest(options)
 		if reqErr != nil {
 			return nil, nil, nil, reqErr
 		}
-
 		log.Infof("Sending CSR (retrial #%d) ...", retries)
-
-		resp, err := c.protocolClient.SendCSR(req, c.platformClient, c.istioCAAddress)
+		
+		resp, err := c.caProtocol.SendCSR(req)
```

Comment:
If the connection is closed, the retries in the loop won't help, right?

Reply:
Good point. Also I've did some survey, seems gRPC can handle connection transient failure by itself, and we can even configure the reconnect backoff setttings, so that the expotential backoff logic are even provided by gRPC already.

Should I just call over gRPC client or keep the our own logic for now and remove it once we are more confident about gRPC's retry logic?

* [Auto reconnect for Go client](https://github.com/grpc/grpc-go/issues/351)
* https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md
* https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
* https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md